csrc/dynamic_transform.cpp

```diff
@@ -761,8 +764,16 @@ void DynamicTransformConcretizer::mutate(TensorDomain* td) {
   // Update the contiguity vector. Drop the contig val if mutated to broadcast
   auto contig = td->contiguity();
 
-  for (const auto i : c10::irange(td->maybeRFactor().size())) {
-    auto original_id = td->maybeRFactor().at(i);
+  const auto& new_alloc = td->hasAllocation() ? alloc_dom
```

Comment:
```suggestion
  const auto& new_maybe_alloc = td->hasAllocation() ? alloc_dom
```

to distinguis it from `alloc_dom`, the real new allocation domain.